src/resultsdisassemblypage.cpp

```diff
@@ -0,0 +1,158 @@
+/*
+  resultsdisassemblypage.cpp
+
+  This file is part of Hotspot, the Qt GUI for performance analysis.
+
+  Copyright (C) 2017-2020 Klarälvdalens Datakonsult AB, a KDAB Group company, info@kdab.com
+  Author: Darya Knysh <d.knysh@nips.ru>
+
+  Licensees holding valid commercial KDAB Hotspot licenses may use this file in
+  accordance with Hotspot Commercial License Agreement provided with the Software.
+
+  Contact info@kdab.com if any conditions of this licensing are not clear to you.
+
+  This program is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 2 of the License, or
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "resultsdisassemblypage.h"
+#include "ui_resultsdisassemblypage.h"
+
+#include <QMenu>
+#include <QFileDialog>
+#include <QFile>
+#include <QTextStream>
+#include <QMessageBox>
+#include <QString>
+#include <QListWidgetItem>
+#include <QProcess>
+#include <QDebug>
+#include <QDir>
+#include <QDirIterator>
+#include <QTemporaryFile>
+
+#include <KRecursiveFilterProxyModel>
+
+#include "parsers/perf/perfparser.h"
+#include "resultsutil.h"
+
+#include "models/costdelegate.h"
+#include "models/hashmodel.h"
+#include "models/topproxy.h"
+#include "models/treemodel.h"
+#include "data.h"
+
+#include <QStandardItemModel>
+
+ResultsDisassemblyPage::ResultsDisassemblyPage(FilterAndZoomStack *filterStack, PerfParser *parser, QWidget *parent)
+        : QWidget(parent)
+        , ui(new Ui::ResultsDisassemblyPage)
+{
+    ui->setupUi(this);
+    ui->splitter->setStretchFactor(0, 1);
+    ui->splitter->setStretchFactor(1, 1);
+    ui->splitter->setStretchFactor(2, 8);
+
+    m_model = new QStandardItemModel();
+}
+
+ResultsDisassemblyPage::~ResultsDisassemblyPage() = default;
+
+void ResultsDisassemblyPage::clear()
+{
+    if (ui->asmView->model() == nullptr)
+        return;
+
+    int rowCount = ui->asmView->model()->rowCount();
+    if (rowCount > 0) {
+        ui->asmView->model()->removeRows(0, rowCount, QModelIndex());
+    }
+}
+
+void ResultsDisassemblyPage::showDisassembly()
+{
+    // Show empty tab when selected symbol is not valid
+    if (m_curSymbol.symbol.isEmpty()) {
+        clear();
+        return;
+    }
+
+    // Call objdump with arguments: addresses range and binary file
+    QString processName = m_objdump;
+    QStringList arguments;
+    arguments << QLatin1String("-d") << QLatin1String("--start-address") << QLatin1String("0x") + QString::number(m_curSymbol.relAddr, 16) <<
```

Comment:
use QStringLiteral here for the string args instead of QLatin1String

but rewrite the `QLatin1String("0x") + QString::number(m_curSymbol.relAddr, 16)` as: `QStringLiteral("0x%1").arg(m_curSymbol.relAddr, 0, 16)`, dito for the relAddr + size arg